tests/Properties/Layout/Tall.hs

```diff
@@ -73,7 +73,7 @@ prop_purelayout_tall n r1 r2 rect = do
       st = fromJust . stack . workspace . current $ x
       ts = pureLayout layout rect st
   return $
-        length ts == length (index x)
+        length ts <= length (index x)
```

Comment:
I noticed that this property which does not holds for new `Tall` layout passes the test. I loosened the condition, but this is too weak. It will be good if there is more fine property for `pureLayout` of `Tall`

Reply:
Yes this is definitely too weak.  You should restore `==` for "normal" values of `tallRatio` and look into what would be a good property for `0` and `1`.

The reasons the tests pass is that it's quite unlikely for a `Rational` to come out as `0` or `1`—perhaps this would even warrant a new type with a custom `Arbitrary` instance.  Manually triggering the test; i.e., writing something like

``` haskell
generate $ prop_purelayout_tall 10 (3/100) 0 (Rectangle 0 0 4 8)
```

will fail all (most) of the time.